src/utils/utils_hetero.py

```diff
@@ -86,36 +138,40 @@ def create_hetero_data_instance(graph, y_bus):
     edge_index['to_bus'] = edge_index['to_bus'].map(index_mapping)
     swapped = deepcopy(edge_index)
     swapped[['from_bus', 'to_bus']] = edge_index[['to_bus', 'from_bus']]
-
-    # Get edge attributes, reset index to match edge_index, and make bidirectional
-    edge_attr = graph.line[['r_ohm_per_km',
-                           'x_ohm_per_km',
-                           'c_nf_per_km',
-                           'g_us_per_km',
-                           'max_i_ka',
-                           'parallel',
-                           'df',
-                           'length_km']].reset_index(drop=True)
+    # edge_index = pd.concat([edge_index, swapped], axis=0).reset_index(drop=True) should we include this line???
+
+    edge_attr_lines_list = []
+    for edges in graph.line.itertuples():
+        # Calculate line admittance from impedance and convert to per-unit system
+        r_tot = float(edges.r_ohm_per_km) * float(edges.length_km)
+        x_tot = float(edges.x_ohm_per_km) * float(edges.length_km)
+        conductance_line, susceptance_line = impedance_to_admittance(r_tot, x_tot, graph.bus['vn_kv'][edges.from_bus], graph.sn_mva)
+        edge_attr_lines_list.append([conductance_line, susceptance_line])
+    
+    # we are missing the swapped edges attributes, to add, create another for loop and uncomment the line above
+    edge_attr = pd.DataFrame(edge_attr_lines_list, columns=['r_tot', 'x_tot'])
```

Comment:
The edge attributes are the same in both directions, so I don't think it's necessary to add the swapped edge attributes, since reverse edges with the same attributes are already added near the end.
Also, the 'r_tot' and 'x_tot' here are actually conductance and susceptance of the line, so it might be nice to rename it.